src/pages/docs/how-to/http-bridge.mdx

```diff
@@ -0,0 +1,392 @@
+but confines them to two separate HTTP headers, `X-Auth-Secret` and `Authorization`, that can be generated
+by clients without interacting with our service.
+
+Details of how to generate the header values can be found in the
+[W3UP UCAN Bridge Protocol](https://github.com/web3-storage/specs/blob/main/w3-ucan-bridge.md) specification, but
+they can also be generated for you by the [`w3` CLI](https://github.com/web3-storage/w3cli):
+
+### Generating Auth Headers with `w3cli`
+
+If you have not already logged in and created a space with the `w3` CLI, do that first:
+
+```sh
+w3 login you@example.com
+# follow onscreen instructions
+w3 space create
+# follow onscreen instructions
+```
+
+Next, find the DID of the space you'd like to use with the bridge:
+
+```sh
+w3 space ls
+* did:key:z6Mkabc123
+```
+
+And use the `bridge generate-tokens` command to generate header values:
+
+```sh
+w3 bridge generate-tokens did:key:z6Mkabc123 --can 'store/add' --can 'upload/add' --can 'upload/list' --expiration `date -v +24H +%s`
+```
+
+The example command above uses the `--expiration` option to specify that these headers will be
+valid for 24 hours. It uses the `--can` options to specify that the headers can be used to invoke
+`store/add`, `upload/add` or `upload/list` capabilities.
+
+The result of `w3 bridge generate-tokens` will contain the values of the `X-Auth-Secret` and `Authorization` headers:
+
+```txt
+X-Auth-Secret header: uOGQyYzRhYWQwNmY3NtEwOTg1ZWU0NDU0NjByNTg2ZGE
+
+Authorization header: uOqJlcm9vdHOB2CpYKQABcRIgYVymlT6sxiDd45CA0f...
+```
+
+These can be used with `curl` like:
+
+```sh
+curl -X POST \
+-H "X-Auth-Secret: uOGQyYzRhYWQwNmY3NtEwOTg1ZWU0NDU0NjByNTg2ZGE" \
+-H "Authorization: uOqJlcm9vdHOB2CpYKQABcRIgYVymlT6sxiDd45CA0f..." \
+...
+https://up.web3.storage/bridge
+```
+
+## Listing Uploads
+
+Listing uploads in a space is one of the easiest ways to ensure the bridge is working. We'll
+give an example using the common unix utility `curl`:
+
+First, create a file named `list.json` and put the following JSON in it, replacing `did:key:z6Mkabc123`
+with the space you used to generate auth headers above:
+
+```json
+{
+  "tasks": [
+    [
+      "upload/list",
+      "did:key:z6Mkabc123",
+      {}
+    ]
+  ]
+}
+```
+
+Next, run `curl` as follows, replacing the values of the `X-Auth-Secret` and `Authorization`
+headers with the values you generated above:
+
+```sh
+curl -X POST \
+-H "X-Auth-Secret: uOGQyYzRhYWQwNmY3NtEwOTg1ZWU0NDU0NjByNTg2ZGE" \
+-H "Authorization: uOqJlcm9vdHOB2CpYKQABcRIgYVymlT6sxiDd45CA0f..." \
+-d @list.json \
+https://up.web3.storage/bridge
+```
+
+## Storing Files
+
+TODO: add information about max upload size (5gb?)
+
+To upload a file to your space you'll first need convert it to a CAR. Assuming your file is
+named `hello.txt` you can first use `ipfs-car` to create a CAR:
+
+```sh
+ipfs-car pack hello.txt > hello.car
+bafybeidhkumeonuwkebh2i4fc7o7lguehauradvlk57gzake6ggjsy372a
+```
```

Comment:
You might also want to add that you can create a CAR from existing IPFS data via:

```console
$ echo "hello world" > hello.txt

$ ipfs add hello.txt --cid-version=1 --wrap-with-directory
added bafkreifjjcie6lypi6ny7amxnfftagclbuxndqonfipmb64f2km2devei4 hello.txt
added bafybeidhkumeonuwkebh2i4fc7o7lguehauradvlk57gzake6ggjsy372a

$ ipfs dag export bafybeidhkumeonuwkebh2i4fc7o7lguehauradvlk57gzake6ggjsy372a > hello.car
```